src/interventions.c

```diff
@@ -736,6 +746,125 @@ void intervention_test_take( model *model, individual *indiv )
+*  Name:		intervention_lateral_flow_test_take
+*  Description: An individual takes a lateral flow test
+*
+*  				At the time of testing it will test positive if the individual has
+*  				sufficient viral load, modeled by their infectiousness.
+*  Returns:		void
+******************************************************************************************/
+void intervention_lateral_flow_test_take( model *model, individual *indiv )
+{
+	if ( indiv->lateral_flow_test_capacity <= 0 ) return;
+	indiv->lateral_flow_test_capacity--;
+
+	int result_time = model->time;
+
+	int time_infected = time_infected( indiv );
+
+	if( time_infected != UNKNOWN )
+	{
+		int infection_type = 0;
+		for( int bucket = 0; bucket < N_NEWLY_INFECTED_STATES; bucket++ )
+		{
+			infection_type = NEWLY_INFECTED_STATES[ bucket ];
+			if( indiv->infection_events->times[ NEWLY_INFECTED_STATES[ bucket ] ] == time_infected )
+				break;
+		}
+
+		time_infected = model->time - time_infected;
+
+		const double infectious_factor = 1/.0802 * exp(1);
+
+		double sensitivity = 0;
+		double I = 0;
+		double V = 0;
+		const int peak_time = model->event_lists[ LATERAL_FLOW_TEST ].infectious_peak_time;
+		const double *infectious_curve = model->event_lists[ infection_type ].infectious_curve[ LATERAL_FLOW_TEST ];
+		const double g = 1.0/8;
+
+		const double b = 1.0/6;
+		if( time_infected <= peak_time )
+		{
+			I = infectious_curve[ time_infected ] * indiv->infectiousness_multiplier * infectious_factor;
+			V = log( I ) / g;
+		}
+		else
+		{
+			I = infectious_curve[ time_infected ] * indiv->infectiousness_multiplier * infectious_factor;
+			V = log( I ) / ( g + b ) + log( infectious_curve[ peak_time ] *
+			                                indiv->infectiousness_multiplier *
+			                                infectious_factor ) * ( 1 / g - 1 / ( g + b ) );
+		}
+
+		if ( V < 0 )
+		{
+			indiv->lateral_flow_test_result = gsl_ran_bernoulli( rng, 1 - model->params->lateral_flow_test_specificity );
+		}
+		else
+		{
+			sensitivity = 1 / ( 1 + exp( -V ) );
+			sensitivity = max( 0, min( model->params->lateral_flow_test_sensitivity , sensitivity ) );
+			indiv->lateral_flow_test_sensitivity = sensitivity;
+			indiv->lateral_flow_test_result = gsl_ran_bernoulli( rng, sensitivity );
+		}
+	}
+	else
+		indiv->lateral_flow_test_result = gsl_ran_bernoulli( rng, 1 - model->params->lateral_flow_test_specificity );
+
+	if ( indiv->lateral_flow_test_result == POSITIVE_TEST )
+		indiv->quarantine_test_result = POSITIVE_TEST;
+
+	if ( indiv->lateral_flow_test_result == POSITIVE_TEST )
+	{
+		indiv->quarantine_test_result = POSITIVE_TEST;
+		add_individual_to_event_list( model, TEST_RESULT, indiv, result_time );
+	}
+	else if ( indiv->lateral_flow_test_capacity > 0 )
+		add_individual_to_event_list( model, LATERAL_FLOW_TEST_TAKE, indiv, model->time + 1 );
+
+}
+
+/*****************************************************************************************
```

Comment:
This function does not seem to be used (unless I've missed it's usage). Please remove.